04_numeronym/jimbotech/numeronym.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"unicode"
+)
+
+var out io.Writer = os.Stdout
+
+// following rules from https://www.revolvy.com/page/Numeronym
+// Program need not convert words with length 3, and should not convert shorter words.
+// If an l (lowercase ell) would occur before a 1 (one), it should be made uppercase.
+// If an I (uppercase eye) would occur before a 1 (one), it should be made lowercase.
+// Input will be printable ASCII with spaces.
+func wordMixer() func(inputStr string) string {
+
+	knownConversions := map[string]string{
+		"year 2038":        "Y2K38",
+		"year 2000":        "Y2K",
+		"i love you":       "143",
+		"eyjafjallajökull": "E15",
+	}
+
+	return func(inputStr string) string {
+
+		inputStr = strings.TrimSpace(inputStr)
+		runes := []rune(inputStr)
+
+		if len(runes) <= 3 {
+			return inputStr
+		}
+
+		lowerRunes := []rune(inputStr)
+		for i, v := range lowerRunes {
```

Comment:
No need for this loop. Just use https://golang.org/pkg/strings/#ToLower:

```
if known, found := knownConversions[strings.ToLower(input)]; found {
        return known
}
```
(i've renamed `value` to `known` because `value` is not very descriptive. If you want to use it, just call it `v`: https://talks.golang.org/2014/names.slide

Reply:
Done